nav2_util/src/node_utils.cpp

```diff
@@ -92,6 +92,10 @@ rclcpp::Node::SharedPtr generate_internal_node(const std::string & prefix)
 
 void setSoftRealTimePriority()
 {
+  #if defined(__APPLE__) || defined(_WIN32)
+    throw std::runtime_error("Cannot set as real-time thread if not on Linux!");
```

Comment:
I'd either like this to implement for Apple / Windows or mention that it is not implemented yet but could be done so in the future with contributions. Otherwise, set the parameter to false.

Reply:
The function currently doesn't have return type, but I guess we could change it to bool and return false on windows/mac and true on Linux. However, it currently already uses throwing a runtime error to signal failure so the proposed change seems more inline with existing code. Then Windows would match the case of Linux failing to set priority.


 We could change the message to 
```
        throw std::runtime_error("Setting priority as real-time thread is currently only supported on Linux. Contributions welcome.");

```